src/utils/havoc.js

```diff
@@ -446,8 +446,34 @@ class ObjectValueHavocingVisitor {
   }
 
   visitAbstractValue(val: AbstractValue): void {
-    for (let i = 0, n = val.args.length; i < n; i++) {
-      this.visitValue(val.args[i]);
+    if (!val.mightBeObject()) {
+      // Only objects need to be havoced.
+      return;
+    }
+    if (val.values.isTop()) {
+      // If we don't know which object instances it might be,
+      // then it might be one of the arguments that created
+      // this value. See #2179.
+
+      // To ensure that we don't forget to provide arguments
+      // that can be havoced, we require at least one argument.
+      let whitelistedKind =
+        val.kind &&
+        (val.kind === "widened numeric property" || // TODO: Widened properties needs to be havocable.
```

Comment:
It seems wrong to for a value that calls out "numeric" in its kind to not have a types domain of NumberValue. Not the fault of this PR, but now that we've noticed....

Reply:
It is the property name that is numeric. Not the value of it.

Reply:
Ah, very misleading.